core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3966,7 +3957,7 @@ object Stream extends StreamLowPriority {
 
               resourceEval {
                 F.delay(init())
-                  .flatMap(i => Algebra.compile(s.get, scope, i)(foldChunk))
+                  .flatMap(i => Algebra.compile(s.get, scope, true, i)(foldChunk))
```

Comment:
When using `s.compile.resource`, we enable suppression of soft scopes (aka scopes that are a result of resource acquisition) for the root scope.

Reply:
Thats an ultra-magic statement :-) I think we would need to clarify it for users.